tcms/core/views.py

```diff
@@ -7,47 +7,50 @@
 from django.db.models import Count, Q
 from django.utils import translation
 from django.utils.translation import trans_real
+from django.utils.decorators import method_decorator
 from django.views.generic.base import View
-from django.views.decorators.http import require_GET
+from django.views.generic.base import TemplateView
 from django.contrib.auth.decorators import login_required
 from django.views.decorators.csrf import requires_csrf_token
 
 from tcms.testplans.models import TestPlan
 from tcms.testruns.models import TestRun
 
 
-@require_GET
-@login_required
-def dashboard(request):  # pylint: disable=missing-permission-required
-    """List all recent TestPlans and TestRuns"""
-    test_plans = TestPlan.objects.filter(
-        author=request.user
-    ).order_by(
-        '-plan_id'
-    ).select_related(
-        'product', 'type'
-    ).annotate(
-        num_runs=Count('run', distinct=True)
-    )
-    test_plans_disable_count = test_plans.filter(is_active=False).count()
-
-    test_runs = TestRun.objects.filter(
-        Q(manager=request.user) |
-        Q(default_tester=request.user) |
-        Q(case_run__assignee=request.user),
-        stop_date__isnull=True,
-    ).order_by('-run_id').distinct()
-
-    context_data = {
-        'test_plans_count': test_plans.count(),
-        'test_plans_disable_count': test_plans_disable_count,
-        'last_15_test_plans': test_plans.filter(is_active=True)[:15],
-
-        'last_15_test_runs': test_runs[:15],
-
-        'test_runs_count': test_runs.count(),
-    }
-    return render(request, 'dashboard.html', context_data)
+@method_decorator(login_required, name='dispatch')
+class DashboardView(TemplateView):
```

Comment:
missing pylint disable,
missing method names

Reply:
Its a TemplateView, and the method names are not necessary in my opinion. It implements only get method. By itself TemplateView shows that only get is expected.

Reply:
FTR specifying the allowed methods in `http_method_names` short-circuits the dispatch logic before returning a 405 response. We can merge this as-is ATM.